x-pack/packages/kbn-elastic-assistant/impl/assistant/api/anonymization_fields/use_fetch_anonymization_fields.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { FindAnonymizationFieldsResponse } from '@kbn/elastic-assistant-common/impl/schemas/anonymization_fields/find_anonymization_fields_route.gen';
+import { HttpSetup } from '@kbn/core/public';
+import { useQuery } from '@tanstack/react-query';
+import {
+  ELASTIC_AI_ASSISTANT_ANONYMIZATION_FIELDS_URL_FIND,
+  ELASTIC_AI_ASSISTANT_API_CURRENT_VERSION,
+} from '@kbn/elastic-assistant-common';
+
+export interface UseFetchAnonymizationFieldsParams {
+  http: HttpSetup;
+  signal?: AbortSignal | undefined;
+}
+
+/**
+ * API call for fetching anonymization fields for current spaceId
+ *
+ * @param {Object} options - The options object.
+ * @param {HttpSetup} options.http - HttpSetup
+ * @param {AbortSignal} [options.signal] - AbortSignal
+ *
+ * @returns {useQuery} hook for getting the status of the anonymization fields
+ */
+export const useFetchAnonymizationFields = ({
+  http,
+  signal,
+}: UseFetchAnonymizationFieldsParams) => {
+  const query = {
+    page: 1,
+    per_page: 100,
```

Comment:
What if there are more than 100 fields?

Reply:
Yeah, I will i crease it to 1000. 
I didn't mentioned that, but on this iteration (till the new design will be ready), I prefer continue to use in-memory EUI table from the initial implementation. So for that reason I'm fetching all the fields and not using the server side paging.